### ISSUE TITLE
JavaScript Comments not being generated in 0.8

### ISSUE THREAD

**Issue by bajmming** [nikhilk/scriptsharp#270]
Hi there,

I've upgraded my Script# project to 0.8 (previously using 0.7.6).

Unfortunately the comments are now not being generated in the JavaScript file.  

The code used to be

public bool EnableDocComments {
            get {
                if (DebugFlavor) {
                    return (_docCommentFile != null);
                }
                return false;
            }
        }

but, is now just 

public bool EnableDocComments {
            get {
                return (_docCommentFile != null);
            }
        }

Is there a Compiler Option or a way I can tell Script# to generate comments in the JavaScript file?

Thanks,

Ben

P.S. I'm really impressed with Script#.  It's been a bit of a tough learning curve due to lack of documentation, but it's very cool :)

**Comment by nikhilk** [comment]
Yes, recently removed the functionality to generate comments into the script. Thats not to say they won't come back. However there were a few questions to visit first -
1. Documentation format - eg. vsdoc vs. jsdoc.
2. The new optimized script generation format for stuff like enums, interfaces doesn't lend well to documenting. The documentation format used to work generally by documenting functions.
3. Correct representation of types in documentation - there were some bugs that needed to be addressed.

Knowing your goals would be interesting input:
- Is it documenting script, so someone looking at the generated script can benefit from documentation. Assuming this is when users of your script aren't using script#, because otherwise the xml documentation would serve better.
- Intellisense for authoring against your script. This is where #2 above gets in the way.
- Any others?

**Comment by bajmming** [comment]
Yes, we're only interested in documenting the script when it's going to be used in other script code, i.e. not using Script#.  I'm currently re-writing our common JavaScript library as a C# project to test how easy it is to use Script#.  This generated script will then be called from other JavaScript.  So, I'd like vsdoc function headers for intellisense and the ability to output comments put in the C# code to help with JavaScript debugging in the browser.  

One of the big advantages that I saw in Script# is the ability to produce an API from the C# code using Sandcastle (or similar) and also the vsdoc comments.  Once I'm happy I can re-produce the common library I'll then look at unit tests and how that can fit in.

So in summary, I'd like to produce vsdoc for Intellisense for when I can't use Script#.  I can produce the API using Sandcastle or something similar.  Also the ability to output comments I've put in the C# code is useful for when I'm debugging in IE as it helps with some of the more complicated functions we have.
